### PR TITLE
Migrate Automatically Check for Updates toggle from About tab to General Tab

### PR DESCRIPTION
**Pull request by Jerry23011**
Please tell me if I CVed the code right :)

### CODE REVIEW

**Review by AkaShark — APPROVED** [review]

**Review by tisfeng — CHANGES_REQUESTED** [review]
I think it would be better to place the AutoCheckUpdate option on the General tab, because it is a common feature for most apps.

The advanced tab should place some less common features,, or features in Easydict that are not well understood by the average user.

image

File: Easydict/App/Localizable.xcstrings [hunk 1]
```diff
@@ -309,6 +309,22 @@
...
+        "en" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Automatically check for application updates (Latest version %@)"
+          }
+        },
+        "zh-Hans" : {
+          "stringUnit" : {
+            "state" : "needs_review",
+            "value" : "自动检查应用程序更新（最新版本 %@）"
```
Comment (tisfeng) [inline Easydict/App/Localizable.xcstrings:323]: The `Latest version` could keep in About tab, we can add a `Check update` button in this page, like 1Password.

This makes it convenient for users to manually check for updates on the About tab, and also provides an alternative way  besides the menu icon.

image

image

**Review by tisfeng — CHANGES_REQUESTED** [review]

File: Easydict/App/Localizable.xcstrings (hunk 1 above)
Comment (Jerry23011) [reply Easydict/App/Localizable.xcstrings:323]: For the check for updates button, I think [CodeEdit](https://github.com/CodeEditApp/CodeEdit/blob/main/CodeEdit/Features/Settings/SettingsView.swift) has a pretty good approach to put it in general
截屏2024-03-06 13 34 58

Comment (tisfeng) [reply Easydict/App/Localizable.xcstrings:323]: Ok, it looks great, let's follow it.

Comment (Jerry23011) [reply Easydict/App/Localizable.xcstrings]: I've updated the code to fit the style of CodeEdit
截屏2024-03-09 00 15 55
截屏2024-03-09 00 17 18

Comment (Jerry23011) [reply Easydict/App/Localizable.xcstrings]: I think we can perhaps rewrite the About view later to have a horizontal layout. It doesn't look very modern at the moment

Comment (tisfeng) [reply Easydict/App/Localizable.xcstrings]: Yes, I agree with you.

File: Easydict/App/Localizable.xcstrings [hunk 2]
```diff
@@ -309,6 +309,22 @@
...
+        "en" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Automatically check for app updates"
+          }
+        },
+        "zh-Hans" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "自动检查应用程序更新"
```
Comment (tisfeng) [inline Easydict/App/Localizable.xcstrings:323]: `auto_check_update` already exists.

image

Comment (Jerry23011) [reply Easydict/App/Localizable.xcstrings:323]: These two cannot be deleted as they are both auto-generated. I check the code and found the previous one to be for the objc About view. I will push a commit to unify their translations.
截屏2024-03-09 14 25 20

**Review by phlpsong — APPROVED** [review]

File: Easydict/App/Localizable.xcstrings (hunk 2 above)
Comment (tisfeng) [reply Easydict/App/Localizable.xcstrings]: ok

**Review by tisfeng — APPROVED** [review]
Looks good.

**Pull request merged**